sympy/physics/control/lti.py

```diff
@@ -802,77 +849,49 @@ def doit(self, **kwargs):
         TransferFunction((2 - s**3)*(-a*p**2 - b*s), (-p + s)*(s**4 + 5*s + 6), s)
 
         """
-        res = None
-        for arg in self.args:
-            arg = arg.doit()
-            if res is None:
-                res = arg
-            else:
-                num_ = arg.num * res.num
-                den_ = arg.den * res.den
-                res = TransferFunction(num_, den_, self.var)
-        return res
+
+        _arg = (arg.doit().to_expr() for arg in self.args)
+        res = Mul(*_arg, evaluate=True)
+        return TransferFunction.from_rational_expression(res, self.var)
 
     def _eval_rewrite_as_TransferFunction(self, *args, **kwargs):
         return self.doit()
 
     def __add__(self, other):
-        if isinstance(other, (TransferFunction, Series)):
-            if not self.var == other.var:
-                raise ValueError("All the transfer functions should use the same complex variable "
-                    "of the Laplace transform.")
+        if not isinstance(other, LinearTimeInvariant):
+             return NotImplemented
 
-            return Parallel(self, other)
-        elif isinstance(other, Parallel):
-            if not self.var == other.var:
-                raise ValueError("All the transfer functions should use the same complex variable "
-                    "of the Laplace transform.")
-            arg_list = list(other.args)
+        if not other.is_SISO:
+            raise TypeError("Cannot add SISO and MIMO systems together.")
 
+        if isinstance(other, Parallel):
+            arg_list = list(other.args)
             return Parallel(self, *arg_list)
-        else:
-            raise ValueError("This transfer function expression is invalid.")
+
+        return Parallel(self, other)
```

Comment:
Maybe this flattening should be handled in `Parallel.__new__`. Are these supposed to be equivalent:
```python
tf1 + tf2 + tf3
Parallel(tf1, Parallel(tf2, tf3))
```